app/invocation/invocation_share_button.tsx

```diff
@@ -12,50 +14,80 @@ import Dialog, {
 import Input from "../components/input/input";
 import Modal from "../components/modal/modal";
 import Select, { Option } from "../components/select/select";
+import rpcService from "../service/rpc_service";
 import InvocationModel from "./invocation_model";
 
 export interface InvocationShareButtonComponentProps {
   user?: User;
   model: InvocationModel;
+  invocationId: string;
 }
 
 interface State {
   isOpen: boolean;
-  // TODO: Replace with request proto once it's available.
-  visibility?: VisibilitySelection;
+  isLoading: boolean;
+  acl: acl.IACL;
+  error?: string;
 }
 
-type VisibilitySelection = "owner" | "group" | "public";
+type VisibilitySelection = "group" | "public";
 
 export default class InvocationShareButtonComponent extends React.Component<
   InvocationShareButtonComponentProps,
   State
 > {
-  state: State = { isOpen: false };
-
   private inputRef = React.createRef<HTMLInputElement>();
 
-  componentDidMount() {
-    // TODO: Fetch current invocation permissions
+  constructor(props: InvocationShareButtonComponentProps) {
+    super(props);
+    this.state = this.getInitialState();
```

Comment:
ubernit: slightly prefer putting this in componentDidMount and removing constructor

that way you don't have to worry about that props stuff

Reply:
I generally prefer using the constructor so that the initial state is what we expect it to be during the first call to `render`. Otherwise you have to handle the possibility of state being null.

Turns out we can still remove the constructor though - looks like TS will auto-generate the constructor for us if we put `this.state = this.getInitialState()` directly in the class. https://bit.ly/3jhdx2J